help/guest-users.md

```diff
@@ -42,6 +42,37 @@ pricing](/help/zulip-cloud-billing#temporary-users-and-guests) for guest users.
 
 {end_tabs}
 
+## Configure whether guests can see all other users
+
+{!cloud-plus-only.md!}
+
+You can restrict guests' ability to see other users in the organization. If you
+do so, guests will be able to see information about other users only in the
+following cases:
+
+- The user belongs to a [direct message](/help/direct-messages) thread with the
+  guest.
+- The user is subscribed to the same [stream](/help/streams-and-topics) as the
+  guest.
+
+All other users will be displayed as **Unknown users** in the Zulip app. For
+example, messages and reactions from a former subscriber of a stream will be
+shown as from an **Unknown user**. Unknown users are not shown in the guest's
+right sidebar, and cannot be accessed by guests via the [API](/api).
+
+{start_tabs}
+
+{tab|desktop-web}
+
+{settings_tab|organization-permissions}
+
+1. Under **Guests**, configure **Who can view all other users in the
+   organization**.
+
+{!save-changes.md!}
+
+{end_tabs}
+
```

Comment:
We need to document this setting somewhere, as one additionally has access to the mapping between user IDs and presence status if it is not enabled, which is the default to avoid creating bad experiences on upgrade to 8.0.

```
# Limiting guest access to other users via the                                                      
# can_access_all_users_group setting makes presence queries much more                               
# expensive. This can be a significant performance problem for                                      
# installations with thousands of users with many guests limited in                                 
# this way, pending further optimization of the relevant code paths.                                
CAN_ACCESS_ALL_USERS_GROUP_LIMITS_PRESENCE = False                                                  
```

We could decide to change that default -- in Zulip Cloud, the feature is limited to just the Plus plan, and I don't think the problematic usage model is likely to be common; meanwhile self-hosters will start with this feature not enabled. So the documentation could say that we recommend caution using this feature in contexts where guests will be subscribed to streams with hundreds of users as it may create load problems for your server, and one can disable the above setting if appropriate.

@alexmv @sahil839 thoughts?

Reply:
What happens in the administrative UI if the setting is not enabled on the server?

Reply:
Oh, I guess it's just IDs -> presence mapping, I see.

Reply:
Opened https://github.com/zulip/zulip/issues/28131 to track this outstanding decision/documentation.